lib/link/connectors.js

```diff
@@ -36,7 +43,7 @@ class WebSocketBaseConnector extends events.EventEmitter {
 	}
 
 	_doHeartbeat() {
-		this._check(["connected"]);
+		this._check(["connected", "closing"]);
 		if (Date.now() - this._lastHeartbeat > 2000 * this._heartbeatInterval) {
```

Comment:
Hm, thinking about this, maybe the heartbeat multiplier - 2000 - used here should be a config option so you can actually configure what the heartbeat timeout you want to use? Some people might prefer having a way smaller timeout interval than a multiplier of 2000 retries...

Reply:
I think you need to pay a bit more attention to the units here.  Date.now() is unix time in milliseconds, this._heartbeatInterval is in seconds.  The logic is saying if the time since the last heartbeat is greater than 2x the heartbeat interval.

Reply:
Can we, maybe, then homogenise the values here to a single base and not mix seconds and microseconds? It either should be microseconds and interval should be defined in those or hide the microseconds and do calculations in seconds as the base.
Personally, I rather just use microseconds overall. Mixing value bases for times is always leading to some kind of WTF later.

Reply:
Would `(Date.now() - this._lastHeartbeat) / 1000 > 2 * this._heartbeatInverval` be acceptable?  This isn't really relevant to this PR though and we can't change JavaScript functions like Date.now(), SetInterval/SetTimeout operating in milliseconds.  Clusterio code like the protocol and config options are in seconds, Prometheus metrics are _strongly_ recommended to be in seconds and I for one do not like configuring things like autosave interval in milliseconds.

Reply:
That works, but to be honest it does read even more convoluted.

Provided we use microseconds for _heartbeatInterval, could be `(Date.now() * 1000) - this.lastHeartbeat > 2 * this._heartbeatInterval` ?